miden-crypto/src/dsa/ecdsa_k256/mod.rs

```diff
@@ -0,0 +1,311 @@
+};
+use rand::{CryptoRng, RngCore};
+use thiserror::Error;
+
+use crate::{
+    Felt, SequentialCommit, StarkField, Word,
+    ecdh::{EphemeralPublicKey, SharedSecret},
+    utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable},
+};
+
+#[cfg(test)]
+mod tests;
+
+// CONSTANTS
+// ================================================================================================
+
+/// Length of secret key in bytes
+const SECRET_KEY_BYTES: usize = 32;
+/// Length of public key in bytes when using compressed format encoding
+pub(crate) const PUBLIC_KEY_BYTES: usize = 33;
+/// Length of signature in bytes using our custom serialization
+const SIGNATURE_BYTES: usize = 66;
+/// Length of signature in bytes using standard serialization i.e., `SEC1`
+const SIGNATURE_STANDARD_BYTES: usize = 64;
+/// Length of scalars for the `secp256k1` curve
+const SCALARS_SIZE_BYTES: usize = 32;
+
+// SECRET KEY
+// ================================================================================================
+
+/// Secret key for ECDSA signature verification over secp256k1 curve.
+pub struct SecretKey {
+    inner: SigningKey,
+}
+
+impl SecretKey {
+    /// Generates a new random secret key using the OS random number generator.
+    #[cfg(feature = "std")]
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        let mut rng = rand::rng();
+
+        Self::with_rng(&mut rng)
+    }
+
+    /// Generates a new secret key using the provided random number generator.
+    pub fn with_rng<R: CryptoRng + RngCore>(rng: &mut R) -> Self {
+        use k256::elliptic_curve::rand_core::SeedableRng;
+
+        let mut seed = [0_u8; 32];
+        rand::RngCore::fill_bytes(rng, &mut seed);
+
+        let mut rng = rand_hc::Hc128Rng::from_seed(seed);
+
+        let signing_key = SigningKey::random(&mut rng);
+        Self { inner: signing_key }
+    }
+
+    /// Gets the corresponding public key for this secret key.
+    pub fn public_key(&self) -> PublicKey {
+        let verifying_key = self.inner.verifying_key();
+        PublicKey { inner: *verifying_key }
+    }
+
+    /// Signs a message (represented as a Word) with this secret key.
+    pub fn sign(&mut self, message: Word) -> Signature {
+        let message_digest = hash_message(message);
+
+        let (signature_inner, recovery_id) = self
+            .inner
+            .sign_prehash_recoverable(&message_digest)
+            .expect("failed to generate signature");
+
+        let (r, s) = signature_inner.split_scalars();
+
+        Signature {
+            r: r.to_bytes().into(),
+            s: s.to_bytes().into(),
+            v: recovery_id.into(),
+        }
+    }
+
+    /// Computes a Diffie-Hellman shared secret from this secret key and the ephemeral public key
+    /// generated by the other party.
+    pub fn get_shared_secret(&self, pk_e: EphemeralPublicKey) -> SharedSecret {
```

Comment:
Given that we can already do `EphemeralSecretKey::diffie_hellman()`, I wonder if we need this method.

Reply:
Not sure, basically the referenced function creates the shared secret on the side of Alice in the example in `ecdh/k256.rs`